CourseSketchProjects/IdentityServer/src/main/java/coursesketch/database/identity/IdentityManager.java

```diff
@@ -0,0 +1,549 @@
+            final String parentId, final Authenticator authChecker)
+            throws DatabaseAccessException, AuthenticationException {
+        final School.ItemType parentType = getParentItemType(itemType);
+        if (!parentType.equals(itemType)) {
+            final AuthenticationResponder responder = authChecker.checkAuthentication(getParentItemType(itemType), parentId, authId, 0,
+                    Authentication.AuthType.newBuilder().setCheckingAdmin(true).build());
+            if (!responder.hasModeratorPermission()) {
+                throw new AuthenticationException("User does not have permission to insert new items for id: " + parentId,
+                        AuthenticationException.INVALID_PERMISSION);
+            }
+        }
+
+        final BasicDBObject insertQuery = createItemInsertQuery(userId, itemId, itemType);
+        if (!parentType.equals(itemType)) {
+            copyParentDetails(insertQuery, itemId, itemType, parentId);
+        }
+
+        // if it is a course
+        if (itemType.equals(School.ItemType.COURSE)) {
+            final String groupId = createNewGroup(userId, itemId);
+            final List<String> groupList = new ArrayList<>();
+            groupList.add(groupId);
+            insertQuery.append(DatabaseStringConstants.USER_LIST, groupList);
+        }
+
+        if (itemType.equals(School.ItemType.BANK_PROBLEM)) {
+            final String groupId = createNewGroup(userId, itemId);
+            final List<String> groupList = new ArrayList<>();
+            groupList.add(groupId);
+            insertQuery.append(DatabaseStringConstants.USER_LIST, groupList);
+
+            if (parentId != null) {
+                insertUserIntoGroup(parentId, groupId, true);
+            }
+        }
+
+        final DBCollection collection = database.getCollection(getCollectionFromType(itemType));
+        collection.insert(insertQuery);
+    }
+
+    /**
+     * Creates a basic query for inserting items into the database.
+     *
+     * @param userId
+     *         The userId of the user that is inserting the new item.
+     * @param itemId
+     *         The id of the item being inserted
+     * @param itemType
+     *         The type of item that is being inserted, EX: {@link School.ItemType#COURSE}
+     * @return {@link BasicDBObject} that contains the basic set up that every item has for its creation.
+     */
+    private BasicDBObject createItemInsertQuery(final String userId, final String itemId, final School.ItemType itemType) {
+        final BasicDBObject query = new BasicDBObject(DatabaseStringConstants.SELF_ID, new ObjectId(itemId));
+        if (School.ItemType.COURSE.equals(itemType)) {
+            query.append(DatabaseStringConstants.COURSE_ID, new ObjectId(itemId))
+                    .append(DatabaseStringConstants.OWNER_ID, userId);
+        }
+        if (School.ItemType.BANK_PROBLEM.equals(itemType)) {
+            query.append(DatabaseStringConstants.PROBLEM_BANK_ID, itemId)
```

Comment:
why are you using 'itemId' alone here and not 'new ObjectId(itemId)' like in like 149?